packages/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx

```diff
@@ -85,7 +85,7 @@ const createScrollCache = (
 });
 type ScrollCache = ReturnType<typeof createScrollCache>;
 
-const isJSDOM = typeof window !== 'undefined' ? /jsdom/.test(window.navigator.userAgent) : false;
+const isJSDOM = typeof navigator !== 'undefined' ? /jsdom/.test(navigator.userAgent) : false;
```

Comment:
This change caused the tests in JSDOM to fail.
Turns out `navigator.userAgent !== window.navigator.userAgent`:

https://github.com/mui/material-ui/blob/c409a0c021df73ba85b6147cdaa1736ade61fd74/packages/test-utils/src/createDOM.js#L57-L58

I'll submit a PR to the core repo to fix this inconsistency.

What was the reason for this change anyway?

Reply:
https://github.com/mui/material-ui/pull/42344

Reply:
It was failing in devmode inside an excel export worker, something being undefined.

Reply:
Not inside a worker but I have no idea why this happens in devmode. I've wrapped in a try.

![Screenshot from 2024-05-22 17-09-32](https://github.com/mui/mui-x/assets/1423607/c1fe0d9d-7256-4e2a-993f-7db0c93f2213)